### PR TITLE
imp: reset conversation only to top of step

### LINKED ISSUE #178: Conversation reset at step
**Describe the solution you'd like**
For for users that are on hold if the developer update the csml flow but don't change the step code in witch the user is on hold it will be better if they continue the conversation instead of resetting the flow and if the developer update the step then de user only need to redo the step and not the flow.

### PR DESCRIPTION
**Pull request by amerelo**
Fixes #178 

## Proposed Changes

- if step code change for users in hold then the conversation rest at the start of the flow
- If step doesn't exist any more then the user conversation is reset at flow start

### CODE REVIEW

**Comment by frsechet** [comment]
@amerelo please rebase after #156 is merged + fix conflicts then ping me for tests 🙂

File: csml_engine/examples/command_line.rs [hunk 1]
```diff
@@ -59,6 +59,7 @@ fn init_bot() -> CsmlBot {
         native_components: Some(load_components().unwrap()),
         custom_components: None,
         default_flow: "flowid".to_owned(),
+        bot_ast: None,
```
Comment (frsechet) [inline csml_engine/examples/command_line.rs:62]: obsolete?

Comment (amerelo) [reply csml_engine/examples/command_line.rs:62]: no i use it for the hold

File: csml_interpreter/examples/bot.rs [hunk 2]
```diff
@@ -92,6 +92,7 @@ fn main() {
         Some(native_components),
         Some(serde_json::json!(custom_components)),
         DEFAULT_FLOW_NAME,
+        None,
```
Comment (frsechet) [inline csml_interpreter/examples/bot.rs:95]: obsolete?

File: csml_interpreter/examples/event.rs [hunk 3]
```diff
@@ -29,6 +29,7 @@ fn main() {
         Some(native_component),
         None,
         DEFAULT_FLOW_NAME,
+        None,
```
Comment (frsechet) [inline csml_interpreter/examples/event.rs:32]: Obsolete?

File: csml_interpreter/examples/factorial.rs [hunk 4]
```diff
@@ -29,6 +29,7 @@ fn main() {
         Some(native_component),
         None,
         DEFAULT_FLOW_NAME,
+        None,
```
Comment (frsechet) [inline csml_interpreter/examples/factorial.rs:32]: Obsolete?

File: csml_interpreter/examples/hello_world.rs [hunk 5]
```diff
@@ -31,6 +31,7 @@ fn main() {
         Some(native_component),
         None,
         DEFAULT_FLOW_NAME,
+        None,
```
Comment (frsechet) [inline csml_interpreter/examples/hello_world.rs:34]: Obsolete?

File: csml_interpreter/examples/memory.rs [hunk 6]
```diff
@@ -29,6 +29,7 @@ fn main() {
         Some(native_component),
         None,
         DEFAULT_FLOW_NAME,
+        None,
```
Comment (frsechet) [inline csml_interpreter/examples/memory.rs:32]: Obsolete?

File: csml_interpreter/examples/metadata.rs [hunk 7]
```diff
@@ -29,6 +29,7 @@ fn main() {
         Some(native_component),
         None,
         DEFAULT_FLOW_NAME,
+        None,
```
Comment (frsechet) [inline csml_interpreter/examples/metadata.rs:32]: Obsolete?

**Pull request merged**